{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "24ca4943-8bb1-476e-87cb-b3e52108aea4",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true
   },
   "source": [
    "# import packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "aa33228c-e2c2-44a0-9b03-213bb35e1bb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(DESeq2)\n",
    "library(data.table)\n",
    "library(dplyr)\n",
    "library(ChIPseeker)\n",
    "library(TxDb.Hsapiens.UCSC.hg38.knownGene)\n",
    "library(EnsDb.Hsapiens.v75)\n",
    "library(clusterProfiler)\n",
    "library(AnnotationDbi)\n",
    "library(XML)\n",
    "library(org.Hs.eg.db)\n",
    "library(ggplot2)\n",
    "library(gridExtra)\n",
    "library(IRdisplay)\n",
    "library(pbapply)\n",
    "library(ChIPpeakAnno)\n",
    "library(sva)\n",
    "library(TCGAbiolinks) \n",
    "library(IRdisplay)\n",
    "library(ggpubr)\n",
    "library(DiffBind)\n",
    "library(Seurat)\n",
    "library(kableExtra)\n",
    "library(tidyverse)\n",
    "library(clusterProfiler)\n",
    "library(enrichplot)\n",
    "library(readxl)\n",
    "library(umap)\n",
    "library(corrplot)\n",
    "library(ggrepel)\n",
    "library(msigdbr)\n",
    "library(ggraph)\n",
    "library(tidygraph)\n",
    "library(GEOquery)\n",
    "library(fgsea)\n",
    "library(ggvenn)\n",
    "library(ComplexHeatmap)\n",
    "library(ggVennDiagram)\n",
    "library(venneuler)\n",
    "library(sva)\n",
    "library(DT)\n",
    "library(tools)\n",
    "library(tidyverse)\n",
    "library(maSigPro)\n",
    "library(RSQLite)\n",
    "\n",
    "library(edgeR)\n",
    "\n",
    "#txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n",
    "\n",
    "options(width = 188)\n",
    "\n",
    "base_dir = '/research_jude/rgs01_jude/groups/jxugrp/home/common/Lab_Members/WenhuoHu/TE/enhancer/'\n",
    "source('~/script/utils.r')\n",
    "setwd(base_dir)\n",
    "\n",
    "path = '/usr/NX/bin:/cm/local/apps/environment-modules/4.5.3//bin:/hpcf/lsf/lsf_prod/10.1/linux3.10-glibc2.17-x86_64/etc:/hpcf/lsf/lsf_prod/10.1/linux3.10-glibc2.17-x86_64/bin:/usr/lpp/mmfs/bin:/usr/lpp/mmfs/lib:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/sbin:/cm/local/apps/environment-modules/4.5.3/bin:/usr/sbin:/sbin:/bin:/usr/bin:/etc:/usr/ucb:/usr/local/bin:/usr/local/local_dfs/bin:/usr/bin/X11:/usr/local/sas:/home/whu78/edirect:/home/whu78/.local/bin:/home/whu78/bin'\n",
    "path = paste0(Sys.getenv('PATH'), ':', path)\n",
    "Sys.setenv(PATH = path)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eea4272b-b626-4b5d-bb9a-3322acf4a66a",
   "metadata": {},
   "source": [
    "# import design table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c3e42152-ddf4-4b57-9e70-eb49da1cb5af",
   "metadata": {},
   "outputs": [],
   "source": [
    "dsn = fread('dsn_sample_atac_may28.tsv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c6f2368a-1fb6-46ce-a1ff-59880d827163",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'base_sra_file'</li><li>'sample_name'</li><li>'cell_type'</li><li>'antibody'</li><li>'srr_id'</li><li>'sample_dir'</li><li>'sra_file'</li><li>'fname'</li><li>'full_fname'</li><li>'ori_dir'</li><li>'cmd'</li><li>'prefetch_cmd'</li><li>'fastqdump_cmd'</li><li>'dump_r'</li><li>'dump_r1'</li><li>'dump_r2'</li><li>'gz_r_cmd'</li><li>'gz_r1_cmd'</li><li>'gz_r2_cmd'</li><li>'r_fastq_gz'</li><li>'r1_fastq_gz'</li><li>'r2_fastq_gz'</li><li>'sede'</li><li>'nn'</li><li>'r1_sample'</li><li>'r2_sample'</li><li>'sample_dir2'</li><li>'r1_fastq_gz_size'</li><li>'r2_fastq_gz_size'</li><li>'r1_sample_size_cal'</li><li>'r2_sample_size_cal'</li><li>'mergeR1_jobname'</li><li>'mergeR1_oo'</li><li>'mergeR1_eo'</li><li>'mergeR1_cmd'</li><li>'mergeR2_jobname'</li><li>'mergeR2_oo'</li><li>'mergeR2_eo'</li><li>'mergeR2_cmd'</li><li>'r1_trimmed'</li><li>'r2_trimmed'</li><li>'cutadapt_jobname'</li><li>'cutadapt_oo'</li><li>'cutadapt_eo'</li><li>'cutadapt_cmd'</li><li>'fname_pre'</li><li>'bow2_jobname'</li><li>'bow2_oo'</li><li>'bow2_eo'</li><li>'sam_file'</li><li>'sam_report_file'</li><li>'bow2_cmd'</li><li>'sam2bam_jobname'</li><li>'sam2bam_oo'</li><li>'sam2bam_eo'</li><li>'bam_file'</li><li>'sam2bam_cmd'</li><li>'bam2bed_jobname'</li><li>'bam2bed_oo'</li><li>'bam2bed_eo'</li><li>'bed_file'</li><li>'bam2bed_cmd'</li><li>'wig_file'</li><li>'fac'</li><li>'fac_n'</li><li>'fac_nn'</li><li>'bed2wig_jobname'</li><li>'bed2wig_oo'</li><li>'bed2wig_eo'</li><li>'bed2wig_cmd'</li><li>'bigwig_file'</li><li>'bed2wig2_jobname'</li><li>'bed2wig2_oo'</li><li>'bed2wig2_eo'</li><li>'bed2wig2_cmd'</li><li>'macs2_jobname'</li><li>'macs2_oo'</li><li>'macs2_eo'</li><li>'macs2_cmd'</li><li>'peak_file'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'base\\_sra\\_file'\n",
       "\\item 'sample\\_name'\n",
       "\\item 'cell\\_type'\n",
       "\\item 'antibody'\n",
       "\\item 'srr\\_id'\n",
       "\\item 'sample\\_dir'\n",
       "\\item 'sra\\_file'\n",
       "\\item 'fname'\n",
       "\\item 'full\\_fname'\n",
       "\\item 'ori\\_dir'\n",
       "\\item 'cmd'\n",
       "\\item 'prefetch\\_cmd'\n",
       "\\item 'fastqdump\\_cmd'\n",
       "\\item 'dump\\_r'\n",
       "\\item 'dump\\_r1'\n",
       "\\item 'dump\\_r2'\n",
       "\\item 'gz\\_r\\_cmd'\n",
       "\\item 'gz\\_r1\\_cmd'\n",
       "\\item 'gz\\_r2\\_cmd'\n",
       "\\item 'r\\_fastq\\_gz'\n",
       "\\item 'r1\\_fastq\\_gz'\n",
       "\\item 'r2\\_fastq\\_gz'\n",
       "\\item 'sede'\n",
       "\\item 'nn'\n",
       "\\item 'r1\\_sample'\n",
       "\\item 'r2\\_sample'\n",
       "\\item 'sample\\_dir2'\n",
       "\\item 'r1\\_fastq\\_gz\\_size'\n",
       "\\item 'r2\\_fastq\\_gz\\_size'\n",
       "\\item 'r1\\_sample\\_size\\_cal'\n",
       "\\item 'r2\\_sample\\_size\\_cal'\n",
       "\\item 'mergeR1\\_jobname'\n",
       "\\item 'mergeR1\\_oo'\n",
       "\\item 'mergeR1\\_eo'\n",
       "\\item 'mergeR1\\_cmd'\n",
       "\\item 'mergeR2\\_jobname'\n",
       "\\item 'mergeR2\\_oo'\n",
       "\\item 'mergeR2\\_eo'\n",
       "\\item 'mergeR2\\_cmd'\n",
       "\\item 'r1\\_trimmed'\n",
       "\\item 'r2\\_trimmed'\n",
       "\\item 'cutadapt\\_jobname'\n",
       "\\item 'cutadapt\\_oo'\n",
       "\\item 'cutadapt\\_eo'\n",
       "\\item 'cutadapt\\_cmd'\n",
       "\\item 'fname\\_pre'\n",
       "\\item 'bow2\\_jobname'\n",
       "\\item 'bow2\\_oo'\n",
       "\\item 'bow2\\_eo'\n",
       "\\item 'sam\\_file'\n",
       "\\item 'sam\\_report\\_file'\n",
       "\\item 'bow2\\_cmd'\n",
       "\\item 'sam2bam\\_jobname'\n",
       "\\item 'sam2bam\\_oo'\n",
       "\\item 'sam2bam\\_eo'\n",
       "\\item 'bam\\_file'\n",
       "\\item 'sam2bam\\_cmd'\n",
       "\\item 'bam2bed\\_jobname'\n",
       "\\item 'bam2bed\\_oo'\n",
       "\\item 'bam2bed\\_eo'\n",
       "\\item 'bed\\_file'\n",
       "\\item 'bam2bed\\_cmd'\n",
       "\\item 'wig\\_file'\n",
       "\\item 'fac'\n",
       "\\item 'fac\\_n'\n",
       "\\item 'fac\\_nn'\n",
       "\\item 'bed2wig\\_jobname'\n",
       "\\item 'bed2wig\\_oo'\n",
       "\\item 'bed2wig\\_eo'\n",
       "\\item 'bed2wig\\_cmd'\n",
       "\\item 'bigwig\\_file'\n",
       "\\item 'bed2wig2\\_jobname'\n",
       "\\item 'bed2wig2\\_oo'\n",
       "\\item 'bed2wig2\\_eo'\n",
       "\\item 'bed2wig2\\_cmd'\n",
       "\\item 'macs2\\_jobname'\n",
       "\\item 'macs2\\_oo'\n",
       "\\item 'macs2\\_eo'\n",
       "\\item 'macs2\\_cmd'\n",
       "\\item 'peak\\_file'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'base_sra_file'\n",
       "2. 'sample_name'\n",
       "3. 'cell_type'\n",
       "4. 'antibody'\n",
       "5. 'srr_id'\n",
       "6. 'sample_dir'\n",
       "7. 'sra_file'\n",
       "8. 'fname'\n",
       "9. 'full_fname'\n",
       "10. 'ori_dir'\n",
       "11. 'cmd'\n",
       "12. 'prefetch_cmd'\n",
       "13. 'fastqdump_cmd'\n",
       "14. 'dump_r'\n",
       "15. 'dump_r1'\n",
       "16. 'dump_r2'\n",
       "17. 'gz_r_cmd'\n",
       "18. 'gz_r1_cmd'\n",
       "19. 'gz_r2_cmd'\n",
       "20. 'r_fastq_gz'\n",
       "21. 'r1_fastq_gz'\n",
       "22. 'r2_fastq_gz'\n",
       "23. 'sede'\n",
       "24. 'nn'\n",
       "25. 'r1_sample'\n",
       "26. 'r2_sample'\n",
       "27. 'sample_dir2'\n",
       "28. 'r1_fastq_gz_size'\n",
       "29. 'r2_fastq_gz_size'\n",
       "30. 'r1_sample_size_cal'\n",
       "31. 'r2_sample_size_cal'\n",
       "32. 'mergeR1_jobname'\n",
       "33. 'mergeR1_oo'\n",
       "34. 'mergeR1_eo'\n",
       "35. 'mergeR1_cmd'\n",
       "36. 'mergeR2_jobname'\n",
       "37. 'mergeR2_oo'\n",
       "38. 'mergeR2_eo'\n",
       "39. 'mergeR2_cmd'\n",
       "40. 'r1_trimmed'\n",
       "41. 'r2_trimmed'\n",
       "42. 'cutadapt_jobname'\n",
       "43. 'cutadapt_oo'\n",
       "44. 'cutadapt_eo'\n",
       "45. 'cutadapt_cmd'\n",
       "46. 'fname_pre'\n",
       "47. 'bow2_jobname'\n",
       "48. 'bow2_oo'\n",
       "49. 'bow2_eo'\n",
       "50. 'sam_file'\n",
       "51. 'sam_report_file'\n",
       "52. 'bow2_cmd'\n",
       "53. 'sam2bam_jobname'\n",
       "54. 'sam2bam_oo'\n",
       "55. 'sam2bam_eo'\n",
       "56. 'bam_file'\n",
       "57. 'sam2bam_cmd'\n",
       "58. 'bam2bed_jobname'\n",
       "59. 'bam2bed_oo'\n",
       "60. 'bam2bed_eo'\n",
       "61. 'bed_file'\n",
       "62. 'bam2bed_cmd'\n",
       "63. 'wig_file'\n",
       "64. 'fac'\n",
       "65. 'fac_n'\n",
       "66. 'fac_nn'\n",
       "67. 'bed2wig_jobname'\n",
       "68. 'bed2wig_oo'\n",
       "69. 'bed2wig_eo'\n",
       "70. 'bed2wig_cmd'\n",
       "71. 'bigwig_file'\n",
       "72. 'bed2wig2_jobname'\n",
       "73. 'bed2wig2_oo'\n",
       "74. 'bed2wig2_eo'\n",
       "75. 'bed2wig2_cmd'\n",
       "76. 'macs2_jobname'\n",
       "77. 'macs2_oo'\n",
       "78. 'macs2_eo'\n",
       "79. 'macs2_cmd'\n",
       "80. 'peak_file'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"base_sra_file\"      \"sample_name\"        \"cell_type\"          \"antibody\"           \"srr_id\"             \"sample_dir\"         \"sra_file\"           \"fname\"             \n",
       " [9] \"full_fname\"         \"ori_dir\"            \"cmd\"                \"prefetch_cmd\"       \"fastqdump_cmd\"      \"dump_r\"             \"dump_r1\"            \"dump_r2\"           \n",
       "[17] \"gz_r_cmd\"           \"gz_r1_cmd\"          \"gz_r2_cmd\"          \"r_fastq_gz\"         \"r1_fastq_gz\"        \"r2_fastq_gz\"        \"sede\"               \"nn\"                \n",
       "[25] \"r1_sample\"          \"r2_sample\"          \"sample_dir2\"        \"r1_fastq_gz_size\"   \"r2_fastq_gz_size\"   \"r1_sample_size_cal\" \"r2_sample_size_cal\" \"mergeR1_jobname\"   \n",
       "[33] \"mergeR1_oo\"         \"mergeR1_eo\"         \"mergeR1_cmd\"        \"mergeR2_jobname\"    \"mergeR2_oo\"         \"mergeR2_eo\"         \"mergeR2_cmd\"        \"r1_trimmed\"        \n",
       "[41] \"r2_trimmed\"         \"cutadapt_jobname\"   \"cutadapt_oo\"        \"cutadapt_eo\"        \"cutadapt_cmd\"       \"fname_pre\"          \"bow2_jobname\"       \"bow2_oo\"           \n",
       "[49] \"bow2_eo\"            \"sam_file\"           \"sam_report_file\"    \"bow2_cmd\"           \"sam2bam_jobname\"    \"sam2bam_oo\"         \"sam2bam_eo\"         \"bam_file\"          \n",
       "[57] \"sam2bam_cmd\"        \"bam2bed_jobname\"    \"bam2bed_oo\"         \"bam2bed_eo\"         \"bed_file\"           \"bam2bed_cmd\"        \"wig_file\"           \"fac\"               \n",
       "[65] \"fac_n\"              \"fac_nn\"             \"bed2wig_jobname\"    \"bed2wig_oo\"         \"bed2wig_eo\"         \"bed2wig_cmd\"        \"bigwig_file\"        \"bed2wig2_jobname\"  \n",
       "[73] \"bed2wig2_oo\"        \"bed2wig2_eo\"        \"bed2wig2_cmd\"       \"macs2_jobname\"      \"macs2_oo\"           \"macs2_eo\"           \"macs2_cmd\"          \"peak_file\"         "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(dsn)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "b2313cc3-66ed-4059-afdc-ab4f8905e30a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "           \n",
       "            ATAC CTCF ERG FLI1 GATA2 H3K27ac H3K27Ac H3K27AC H3K27me3 H3K4me3 HiC IgG LMO2 LYL1 Pol2 PU.1 PU1 RNAseq RUNX1 STAG2 TAL1\n",
       "  CMP          0    3   3    3     3       1       0       2        1       1   2   1    3    3    2    0   3      0     3     2    3\n",
       "  GMP          0    3   3    3     3       1       0       2        1       1   2   1    3    3    2    0   3      0     3     2    3\n",
       "  HSC          0    3   3    3     3       1       0       2        1       1   2   1    3    3    2    0   3      0     3     2    3\n",
       "  Kasumi1      1    0   1    1     1       0       1       0        0       0   0   0    1    1    0    1   0      0     1     0    1\n",
       "  KG1          1    0   1    1     1       0       1       0        0       0   0   1    1    1    0    1   0      0     1     0    1\n",
       "  ME1          1    0   1    1     1       0       1       0        0       0   0   1    1    1    0    1   0      0     1     0    1\n",
       "  MEP          0    3   3    3     3       1       0       2        1       1   2   1    3    3    2    0   3      0     3     2    3\n",
       "  TSU1621MT    0    0   1    1     1       0       0       0        0       0   0   0    1    1    0    1   0      1     1     0    1"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(dsn$cell_type, dsn$antibody)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "feb24R",
   "language": "R",
   "name": "feb24r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
