{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "39147a37-cf84-4d2d-be61-3d03fcd7ddb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(DESeq2)\n",
    "library(data.table)\n",
    "library(dplyr)\n",
    "library(ChIPseeker)\n",
    "library(TxDb.Hsapiens.UCSC.hg38.knownGene)\n",
    "library(EnsDb.Hsapiens.v75)\n",
    "library(clusterProfiler)\n",
    "library(AnnotationDbi)\n",
    "library(org.Hs.eg.db)\n",
    "library(ggplot2)\n",
    "library(gridExtra)\n",
    "library(IRdisplay)\n",
    "library(ChIPpeakAnno)\n",
    "library(IRdisplay)\n",
    "library(ggpubr)\n",
    "library(DiffBind)\n",
    "library(kableExtra)\n",
    "library(tidyverse)\n",
    "library(readxl)\n",
    "library(umap)\n",
    "library(ggrepel)\n",
    "library(msigdbr)\n",
    "library(fgsea)\n",
    "library(ggvenn)\n",
    "library(ComplexHeatmap)\n",
    "library(ggVennDiagram)\n",
    "library(venneuler)\n",
    "library(rjson)\n",
    "library(sva)\n",
    "library(DT)\n",
    "\n",
    "library(edgeR)\n",
    "\n",
    "txdb <- TxDb.Hsapiens.UCSC.hg38.knownGene\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c14b5d68-6495-44cd-b2d7-52845bb93fd4",
   "metadata": {},
   "outputs": [],
   "source": [
    "source('~/script/utils.r')\n",
    "setwd('/research_jude/rgs01_jude/groups/jxugrp/home/common/Lab_Members/WenhuoHu/collab_Farhan//')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "4a699256-15ff-40e9-b4f0-4f14c1ec9618",
   "metadata": {},
   "outputs": [],
   "source": [
    "base_dir = '/research_jude/rgs01_jude/groups/jxugrp/home/common/Data_Public/RNA_seq/Blueprint_EGA_Data_EGAD00001000745_Transcriptional_Diversity_Human_Blood_progenitors_Chen_etal_Science_2014/' \n",
    "ega_df = fread('data/ega_list.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6932c340-f8bf-4af6-ba6c-004a41bfa5a8",
   "metadata": {},
   "source": [
    "# download EGA meta info analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b3a9d281-a9dd-46bb-bf10-d31b2273631d",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "   --url https://metadata.ega-archive.org/analyses \\\n",
    "   --header 'Accept: ' \\\n",
    "   --header 'Authorization: Bearer undefined' \\\n",
    "   --header 'Content-Type: application/json' > ega_analysis\n",
    "# 275 M"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "51097438-f277-4baf-b131-9624a51429e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/datasets \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7f6365e2-f019-45e7-8481-488f13cbe313",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/studies \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_studies\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "536ac067-4029-4adf-a178-2667d6ad1872",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/samples \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_samples\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70675af5-36dc-41d0-ba8d-61ba01e5c4bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/experiments \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_experiments\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1925dd8-67ea-473b-8ffa-ff2bf9196e71",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/runs \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_runs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f52f1d7-5859-4a73-acef-ba7140c0f380",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/files/accession_id \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > ega_files\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bba74db6-327e-4008-accc-8f6eb6d25787",
   "metadata": {},
   "source": [
    "# import datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "aa5e2286-0da0-4d67-926e-4914b25500d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_analysis_dt =  jsonlite::fromJSON( 'ega_analysis' )\n",
    "saveRDS(ega_analysis_dt, file = 'data/ega_analysis_dt.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "633fc2ae-3858-495c-81f1-e46384ecf4d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_datasets_dt =  jsonlite::fromJSON( 'ega_datasets' )\n",
    "saveRDS(ega_datasets_dt, file = 'data/ega_datasets_dt.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "c0596b9b-1e05-437f-95d0-6a45f8429967",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_studies_dt =  jsonlite::fromJSON( 'ega_studies' )\n",
    "saveRDS(ega_studies_dt, file = 'data/ega_studies_dt.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "e2925ab5-6919-4e95-9909-022e6f551f2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_samples_dt =  jsonlite::fromJSON( 'ega_samples' )\n",
    "saveRDS(ega_samples_dt, file = 'data/ega_samples_dt.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "id": "65f88674-3025-44a7-abe5-dbcc8eef08ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_experiments_dt =  jsonlite::fromJSON( 'ega_experiments' )\n",
    "saveRDS(ega_experiments_dt, file = 'data/ega_experiments_dt.rds')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ead93c5e-5697-446a-bbb8-356980004696",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/datasets/EGAD00001000745/mappings/study_experiment_run_sample \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json' > a.json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b048a742-16f4-439b-9dc3-75217f01f4fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/files/EGAD00001000745 \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e2e5d61-1283-4c0b-b373-366023367fba",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/datasets/EGAD00001000745/mappings/sample_file \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6e0cb9ca-6d66-4d72-b3ad-210ff235bd3c",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_datasets_dt =  readRDS(file = 'data/ega_datasets_dt.rds')\n",
    "ega_datasets_dt = as.data.table(ega_datasets_dt)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "99d3761c-f4fa-4a24-846a-bdd632adafc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'list'"
      ],
      "text/latex": [
       "'list'"
      ],
      "text/markdown": [
       "'list'"
      ],
      "text/plain": [
       "[1] \"list\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "class(sel$dataset_types)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "1349d49f-9733-46c8-be99-38ffcc817d7b",
   "metadata": {},
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "Error in match.arg(useNA): 'arg' must be NULL or a character vector\n",
     "output_type": "error",
     "traceback": [
      "Error in match.arg(useNA): 'arg' must be NULL or a character vector\nTraceback:\n",
      "1. table(unlist(sel$dataset_types), useNA = T)",
      "2. standardGeneric(\"table\")",
      "3. eval(mc, env)",
      "4. eval(mc, env)",
      "5. eval(mc, env)",
      "6. table(...)",
      "7. base::table(...)",
      "8. match.arg(useNA)",
      "9. stop(\"'arg' must be NULL or a character vector\")"
     ]
    }
   ],
   "source": [
    "sel = ega_datasets_dt[num_samples > 0, ]\n",
    "sel = sel[grep('Blue', title, ignore.case = T),]\n",
    "table(unlist(sel$dataset_types), useNA = T)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "91828f3f-44ef-451f-a038-60b98e916419",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 4 Ã— 12</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>dataset_types</th><th scope=col>technologies</th><th scope=col>num_samples</th><th scope=col>access_type</th><th scope=col>is_in_beacon</th><th scope=col>is_released</th><th scope=col>released_date</th><th scope=col>is_deprecated</th><th scope=col>policy_accession_id</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>EGAD00001000745</td><td>BLUEPRINT: RNA-seq of progenitor cells                                   </td><td>Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                              </td><td>Transcri....</td><td>Illumina....</td><td>26</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><td>EGAD00001001871</td><td>BLUEPRINT Epigenetic characterization of megakaryocytes and erythroblasts</td><td>Megakaryocytes and erythroblasts derive from the same progenitor cell type but carry out very different functions. In order to understand how the different functional phenotypes arise we have characterised the epigenetic landscape of these cells.</td><td>Transcri....</td><td>Illumina....</td><td>20</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2001-01-01T01:00:00+01:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><td>EGAD00001002732</td><td>BLUEPRINT DNA methylation profiles of human hematopoietic progenitors    </td><td>DNA methylation was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, MEP, GMP, CLP, MLP0, MLP1, MLP2, MLP3, MK, CD4+ Tcell, CD8+ Tcell, Bcell, NK, Neut, Mono).            </td><td>Methylat....</td><td>Illumina....</td><td>63</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-10-12T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><td>EGAD00001002733</td><td>BLUEPRINT Gene expression profiles of human hematopoietic progenitors    </td><td>Gene expression was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, GMP, CLP, MLP0, MLP1, MLP2, MLP3).                                                                    </td><td>Transcri....</td><td>Illumina....</td><td>13</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-10-12T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 4 Ã— 12\n",
       "\\begin{tabular}{llllllllllll}\n",
       " accession\\_id & title & description & dataset\\_types & technologies & num\\_samples & access\\_type & is\\_in\\_beacon & is\\_released & released\\_date & is\\_deprecated & policy\\_accession\\_id\\\\\n",
       " <chr> & <chr> & <chr> & <list> & <list> & <int> & <chr> & <lgl> & <lgl> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t EGAD00001000745 & BLUEPRINT: RNA-seq of progenitor cells                                    & Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                               & Transcri.... & Illumina.... & 26 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t EGAD00001001871 & BLUEPRINT Epigenetic characterization of megakaryocytes and erythroblasts & Megakaryocytes and erythroblasts derive from the same progenitor cell type but carry out very different functions. In order to understand how the different functional phenotypes arise we have characterised the epigenetic landscape of these cells. & Transcri.... & Illumina.... & 20 & controlled & FALSE & TRUE & 2001-01-01T01:00:00+01:00 & FALSE & EGAP00001000142\\\\\n",
       "\t EGAD00001002732 & BLUEPRINT DNA methylation profiles of human hematopoietic progenitors     & DNA methylation was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, MEP, GMP, CLP, MLP0, MLP1, MLP2, MLP3, MK, CD4+ Tcell, CD8+ Tcell, Bcell, NK, Neut, Mono).             & Methylat.... & Illumina.... & 63 & controlled & FALSE & TRUE & 2016-10-12T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t EGAD00001002733 & BLUEPRINT Gene expression profiles of human hematopoietic progenitors     & Gene expression was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, GMP, CLP, MLP0, MLP1, MLP2, MLP3).                                                                     & Transcri.... & Illumina.... & 13 & controlled & FALSE & TRUE & 2016-10-12T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 4 Ã— 12\n",
       "\n",
       "| accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | dataset_types &lt;list&gt; | technologies &lt;list&gt; | num_samples &lt;int&gt; | access_type &lt;chr&gt; | is_in_beacon &lt;lgl&gt; | is_released &lt;lgl&gt; | released_date &lt;chr&gt; | is_deprecated &lt;lgl&gt; | policy_accession_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| EGAD00001000745 | BLUEPRINT: RNA-seq of progenitor cells                                    | Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                               | Transcri.... | Illumina.... | 26 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| EGAD00001001871 | BLUEPRINT Epigenetic characterization of megakaryocytes and erythroblasts | Megakaryocytes and erythroblasts derive from the same progenitor cell type but carry out very different functions. In order to understand how the different functional phenotypes arise we have characterised the epigenetic landscape of these cells. | Transcri.... | Illumina.... | 20 | controlled | FALSE | TRUE | 2001-01-01T01:00:00+01:00 | FALSE | EGAP00001000142 |\n",
       "| EGAD00001002732 | BLUEPRINT DNA methylation profiles of human hematopoietic progenitors     | DNA methylation was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, MEP, GMP, CLP, MLP0, MLP1, MLP2, MLP3, MK, CD4+ Tcell, CD8+ Tcell, Bcell, NK, Neut, Mono).             | Methylat.... | Illumina.... | 63 | controlled | FALSE | TRUE | 2016-10-12T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| EGAD00001002733 | BLUEPRINT Gene expression profiles of human hematopoietic progenitors     | Gene expression was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, GMP, CLP, MLP0, MLP1, MLP2, MLP3).                                                                     | Transcri.... | Illumina.... | 13 | controlled | FALSE | TRUE | 2016-10-12T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id   \n",
       "1 EGAD00001000745\n",
       "2 EGAD00001001871\n",
       "3 EGAD00001002732\n",
       "4 EGAD00001002733\n",
       "  title                                                                    \n",
       "1 BLUEPRINT: RNA-seq of progenitor cells                                   \n",
       "2 BLUEPRINT Epigenetic characterization of megakaryocytes and erythroblasts\n",
       "3 BLUEPRINT DNA methylation profiles of human hematopoietic progenitors    \n",
       "4 BLUEPRINT Gene expression profiles of human hematopoietic progenitors    \n",
       "  description                                                                                                                                                                                                                                           \n",
       "1 Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                              \n",
       "2 Megakaryocytes and erythroblasts derive from the same progenitor cell type but carry out very different functions. In order to understand how the different functional phenotypes arise we have characterised the epigenetic landscape of these cells.\n",
       "3 DNA methylation was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, MEP, GMP, CLP, MLP0, MLP1, MLP2, MLP3, MK, CD4+ Tcell, CD8+ Tcell, Bcell, NK, Neut, Mono).            \n",
       "4 Gene expression was analyzed for stem/progenitor cell types and terminally differentiated cell types of the human blood lineage (HSC, MPP, CMP, GMP, CLP, MLP0, MLP1, MLP2, MLP3).                                                                    \n",
       "  dataset_types technologies num_samples access_type is_in_beacon is_released\n",
       "1 Transcri....  Illumina.... 26          controlled  FALSE        TRUE       \n",
       "2 Transcri....  Illumina.... 20          controlled  FALSE        TRUE       \n",
       "3 Methylat....  Illumina.... 63          controlled  FALSE        TRUE       \n",
       "4 Transcri....  Illumina.... 13          controlled  FALSE        TRUE       \n",
       "  released_date             is_deprecated policy_accession_id\n",
       "1 2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "2 2001-01-01T01:00:00+01:00 FALSE         EGAP00001000142    \n",
       "3 2016-10-12T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "4 2016-10-12T02:00:00+02:00 FALSE         EGAP00001000142    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#ega_datasets_dt[num_samples > 10, ][grep('blue', description, ignore.case = T), ]\n",
    "ega_datasets_dt[num_samples > 10, ][grep('blue', title, ignore.case = T), ][grep('progenitor', description, ignore.case = T), ]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0742a78-3f85-4272-9497-5a88dbc91df6",
   "metadata": {},
   "source": [
    "# out meta file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3426ac2-1407-4bd5-8d8a-02678ef8daf9",
   "metadata": {},
   "outputs": [],
   "source": [
    "token = 'eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJ3bF9hQkNXY0oyaTJMV0tZVnJFeFpCazBySnBxeGo3YVZ5YURyZC1qRFV3In0.eyJleHAiOjE3MTQwNzQ1NjQsImlhdCI6MTcxNDA3NDI2NCwianRpIjoiMWQxMTRhNTctZmVmMy00NzJlLTg2NDktZGQ0YmRkODkxMGQ1IiwiaXNzIjoiaHR0cHM6Ly9pZHAuZWdhLWFyY2hpdmUub3JnL3JlYWxtcy9FR0EiLCJzdWIiOiJmOjU2YjZkN2U4LTEyYWEtNGM2Ni1iZTE2LThiODVhMTI0ZTJlNTo0NTUwMiIsInR5cCI6IkJlYXJlciIsImF6cCI6Im1ldGFkYXRhLWFwaSIsInNlc3Npb25fc3RhdGUiOiI3MDlkZTk3ZS0wNmRkLTRjZTQtOGQ3Yi00MTJiNzdmZWFjNmUiLCJzY29wZSI6IiIsInNpZCI6IjcwOWRlOTdlLTA2ZGQtNGNlNC04ZDdiLTQxMmI3N2ZlYWM2ZSIsImVnYV9pZCI6NDU1MDJ9.yEWCvrZGx4sbc5gFTKXUlfLpXFIVQ65OqjAOIxC7mBl5inCyny'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c90acdc9-5430-4583-9d12-002474f40c47",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl https://metadata.ega-archive.org/datasets/{datasetID}/mappings/study_experiment_run_sample \\\n",
    "-H 'Authorization: Bearer access_token'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "acfcaaee-95ce-41e6-8a9e-07f57f63f67a",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/datasets/EGAD00001000745/mappings/sample_file \\\n",
    "  -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJ3bF9hQkNXY0oyaTJMV0tZVnJFeFpCazBySnBxeGo3YVZ5YURyZC1qRFV3In0.eyJleHAiOjE3MTQwNzQ1NjQsImlhdCI6MTcxNDA3NDI2NCwianRpIjoiMWQxMTRhNTctZmVmMy00NzJlLTg2NDktZGQ0YmRkODkxMGQ1IiwiaXNzIjoiaHR0cHM6Ly9pZHAuZWdhLWFyY2hpdmUub3JnL3JlYWxtcy9FR0EiLCJzdWIiOiJmOjU2YjZkN2U4LTEyYWEtNGM2Ni1iZTE2LThiODVhMTI0ZTJlNTo0NTUwMiIsInR5cCI6IkJlYXJlciIsImF6cCI6Im1ldGFkYXRhLWFwaSIsInNlc3Npb25fc3RhdGUiOiI3MDlkZTk3ZS0wNmRkLTRjZTQtOGQ3Yi00MTJiNzdmZWFjNmUiLCJzY29wZSI6IiIsInNpZCI6IjcwOWRlOTdlLTA2ZGQtNGNlNC04ZDdiLTQxMmI3N2ZlYWM2ZSIsImVnYV9pZCI6NDU1MDJ9.yEWCvrZGx4sbc5gFTKXUlfLpXFIVQ65OqjAOIxC7mBl5inCyny' \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "19cab6a3-c949-4bce-8698-490b94236137",
   "metadata": {},
   "outputs": [],
   "source": [
    "ega_dt = fread('data/ega_list.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "3566c38a-916c-4267-865e-2bb23d44cd42",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 Ã— 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>Dataset ID</th><th scope=col>Description</th><th scope=col>Technology</th><th scope=col>Samples</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>EGAD00001000745</td><td>Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                           </td><td>Illumina HiSeq 2000Â PacBio RS</td><td>26</td></tr>\n",
       "\t<tr><td>EGAD00001000903</td><td>RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811</td><td>Illumina HiSeq 2000          </td><td> 4</td></tr>\n",
       "\t<tr><td>EGAD00001000907</td><td>RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                      </td><td>Illumina HiSeq 2000          </td><td> 3</td></tr>\n",
       "\t<tr><td>EGAD00001000911</td><td>RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                  </td><td>Illumina HiSeq 2000          </td><td> 4</td></tr>\n",
       "\t<tr><td>EGAD00001000915</td><td>RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                        </td><td>Illumina HiSeq 2000          </td><td> 4</td></tr>\n",
       "\t<tr><td>EGAD00001000918</td><td>RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                    </td><td>Illumina HiSeq 2000          </td><td> 3</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 Ã— 4\n",
       "\\begin{tabular}{llll}\n",
       " Dataset ID & Description & Technology & Samples\\\\\n",
       " <chr> & <chr> & <chr> & <int>\\\\\n",
       "\\hline\n",
       "\t EGAD00001000745 & Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                            & Illumina HiSeq 2000Â PacBio RS & 26\\\\\n",
       "\t EGAD00001000903 & RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811 & Illumina HiSeq 2000           &  4\\\\\n",
       "\t EGAD00001000907 & RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                       & Illumina HiSeq 2000           &  3\\\\\n",
       "\t EGAD00001000911 & RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                                   & Illumina HiSeq 2000           &  4\\\\\n",
       "\t EGAD00001000915 & RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                         & Illumina HiSeq 2000           &  4\\\\\n",
       "\t EGAD00001000918 & RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                     & Illumina HiSeq 2000           &  3\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 Ã— 4\n",
       "\n",
       "| Dataset ID &lt;chr&gt; | Description &lt;chr&gt; | Technology &lt;chr&gt; | Samples &lt;int&gt; |\n",
       "|---|---|---|---|\n",
       "| EGAD00001000745 | Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                            | Illumina HiSeq 2000Â PacBio RS | 26 |\n",
       "| EGAD00001000903 | RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811 | Illumina HiSeq 2000           |  4 |\n",
       "| EGAD00001000907 | RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                       | Illumina HiSeq 2000           |  3 |\n",
       "| EGAD00001000911 | RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                   | Illumina HiSeq 2000           |  4 |\n",
       "| EGAD00001000915 | RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                         | Illumina HiSeq 2000           |  4 |\n",
       "| EGAD00001000918 | RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                     | Illumina HiSeq 2000           |  3 |\n",
       "\n"
      ],
      "text/plain": [
       "  Dataset ID     \n",
       "1 EGAD00001000745\n",
       "2 EGAD00001000903\n",
       "3 EGAD00001000907\n",
       "4 EGAD00001000911\n",
       "5 EGAD00001000915\n",
       "6 EGAD00001000918\n",
       "  Description                                                                                                                                                                                                                                                                                                                        \n",
       "1 Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                           \n",
       "2 RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811\n",
       "3 RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                      \n",
       "4 RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                  \n",
       "5 RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                        \n",
       "6 RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                    \n",
       "  Technology                    Samples\n",
       "1 Illumina HiSeq 2000Â PacBio RS 26     \n",
       "2 Illumina HiSeq 2000            4     \n",
       "3 Illumina HiSeq 2000            3     \n",
       "4 Illumina HiSeq 2000            4     \n",
       "5 Illumina HiSeq 2000            4     \n",
       "6 Illumina HiSeq 2000            3     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ega_dt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "80b3faf2-2e31-4dc0-8386-9d927e0a1bbd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000745'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000903'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000907'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000911'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000915'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000918'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000919'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000922'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000939'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001140'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001146'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001156'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001165'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001169'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001177'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001184'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001186'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001492'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001501'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001515'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001534'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001538'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001550'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001558'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001561'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002306'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002316'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002339'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002358'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002363'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002433'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002478'</li><li>'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002489'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000745'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000903'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000907'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000911'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000915'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000918'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000919'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000922'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001000939'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001140'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001146'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001156'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001165'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001169'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001177'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001184'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001186'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001492'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001501'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001515'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001534'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001538'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001550'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001558'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001001561'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002306'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002316'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002339'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002358'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002363'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002433'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002478'\n",
       "\\item 'pyega3 -cf credential\\_file.json -c 5 fetch EGAD00001002489'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000745'\n",
       "2. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000903'\n",
       "3. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000907'\n",
       "4. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000911'\n",
       "5. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000915'\n",
       "6. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000918'\n",
       "7. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000919'\n",
       "8. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000922'\n",
       "9. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001000939'\n",
       "10. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001140'\n",
       "11. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001146'\n",
       "12. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001156'\n",
       "13. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001165'\n",
       "14. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001169'\n",
       "15. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001177'\n",
       "16. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001184'\n",
       "17. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001186'\n",
       "18. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001492'\n",
       "19. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001501'\n",
       "20. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001515'\n",
       "21. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001534'\n",
       "22. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001538'\n",
       "23. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001550'\n",
       "24. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001558'\n",
       "25. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001001561'\n",
       "26. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002306'\n",
       "27. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002316'\n",
       "28. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002339'\n",
       "29. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002358'\n",
       "30. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002363'\n",
       "31. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002433'\n",
       "32. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002478'\n",
       "33. 'pyega3 -cf credential_file.json -c 5 fetch EGAD00001002489'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000745\"\n",
       " [2] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000903\"\n",
       " [3] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000907\"\n",
       " [4] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000911\"\n",
       " [5] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000915\"\n",
       " [6] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000918\"\n",
       " [7] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000919\"\n",
       " [8] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000922\"\n",
       " [9] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001000939\"\n",
       "[10] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001140\"\n",
       "[11] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001146\"\n",
       "[12] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001156\"\n",
       "[13] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001165\"\n",
       "[14] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001169\"\n",
       "[15] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001177\"\n",
       "[16] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001184\"\n",
       "[17] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001186\"\n",
       "[18] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001492\"\n",
       "[19] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001501\"\n",
       "[20] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001515\"\n",
       "[21] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001534\"\n",
       "[22] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001538\"\n",
       "[23] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001550\"\n",
       "[24] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001558\"\n",
       "[25] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001001561\"\n",
       "[26] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002306\"\n",
       "[27] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002316\"\n",
       "[28] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002339\"\n",
       "[29] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002358\"\n",
       "[30] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002363\"\n",
       "[31] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002433\"\n",
       "[32] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002478\"\n",
       "[33] \"pyega3 -cf credential_file.json -c 5 fetch EGAD00001002489\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "cmd = paste0('pyega3 -cf credential_file.json -c 5 fetch ', ega_dt$`Dataset ID`)\n",
    "cmd\n",
    "write(cmd, file = '/research_jude/rgs01_jude/groups/jxugrp/home/common/Data_Public/RNA_seq/Blueprint_EGA_Data_EGAD00001000745_Transcriptional_Diversity_Human_Blood_progenitors_Chen_etal_Science_2014/fastq/ega._all.sh')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53b3a74b-ac00-4a8b-9138-f79ae9acaed8",
   "metadata": {},
   "outputs": [],
   "source": [
    "'ega_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "28e36dcf-879f-43ab-9339-90d563914adb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 10</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>abstract</th><th scope=col>study_type</th><th scope=col>pubmed_ids</th><th scope=col>external_links</th><th scope=col>is_released</th><th scope=col>released_date</th><th scope=col>is_deprecated</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>DUMMY_STUDY    </td><td>q                                                                                       </td><td>NA                                                                                                                                                                                                                        </td><td>NA</td><td>Other</td><td></td><td>NA</td><td>FALSE</td><td>2016-09-26T10:49:02+02:00</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>EGAS00000000001</td><td>WTCCC case-control study for Bipolar Disorder                                           </td><td>WTCCC genome-wide case-control association study for Bipolar disorder (BD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                 </td><td>NA</td><td>Other</td><td></td><td>NA</td><td> TRUE</td><td>1980-01-01T02:00:00+01:00</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>EGAS00000000002</td><td>WTCCC case-control study for Bipolar Disorder - Combined Controls                       </td><td>WTCCC genome-wide case-control association study for Bipolar Disorder (BD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.           </td><td>NA</td><td>Other</td><td></td><td>NA</td><td> TRUE</td><td>1980-01-01T02:00:00+01:00</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>EGAS00000000003</td><td>WTCCC case-control study for Coronary Artery Disease                                    </td><td>WTCCC genome-wide case-control association study for Bipolar disorder (CAD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                </td><td>NA</td><td>Other</td><td></td><td>NA</td><td> TRUE</td><td>1980-01-01T02:00:00+01:00</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>EGAS00000000004</td><td>WTCCC case-control study for Coronary Artery Disease - Combined Controls                </td><td>WTCCC genome-wide case-control association study for Coronary Artery Disease (CAD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.   </td><td>NA</td><td>Other</td><td></td><td>NA</td><td> TRUE</td><td>1980-01-01T02:00:00+01:00</td><td>FALSE</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>EGAS00000000005</td><td>WTCCC case-control study for Coronary Artery Disease, Hypertension, T2D - combined cases</td><td>WTCCC genome-wide case-control association study using the cardiovascular disease CAD, HT and T2D as combined case collection and the 1958 British Birth Cohort and the UK National Blood Service collections as controls.</td><td>NA</td><td>Other</td><td></td><td>NA</td><td> TRUE</td><td>1980-01-01T02:00:00+01:00</td><td>FALSE</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 10\n",
       "\\begin{tabular}{r|llllllllll}\n",
       "  & accession\\_id & title & description & abstract & study\\_type & pubmed\\_ids & external\\_links & is\\_released & released\\_date & is\\_deprecated\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <list> & <lgl> & <lgl> & <chr> & <lgl>\\\\\n",
       "\\hline\n",
       "\t1 & DUMMY\\_STUDY     & q                                                                                        & NA                                                                                                                                                                                                                         & NA & Other &  & NA & FALSE & 2016-09-26T10:49:02+02:00 & FALSE\\\\\n",
       "\t2 & EGAS00000000001 & WTCCC case-control study for Bipolar Disorder                                            & WTCCC genome-wide case-control association study for Bipolar disorder (BD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                  & NA & Other &  & NA &  TRUE & 1980-01-01T02:00:00+01:00 & FALSE\\\\\n",
       "\t3 & EGAS00000000002 & WTCCC case-control study for Bipolar Disorder - Combined Controls                        & WTCCC genome-wide case-control association study for Bipolar Disorder (BD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.            & NA & Other &  & NA &  TRUE & 1980-01-01T02:00:00+01:00 & FALSE\\\\\n",
       "\t4 & EGAS00000000003 & WTCCC case-control study for Coronary Artery Disease                                     & WTCCC genome-wide case-control association study for Bipolar disorder (CAD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                 & NA & Other &  & NA &  TRUE & 1980-01-01T02:00:00+01:00 & FALSE\\\\\n",
       "\t5 & EGAS00000000004 & WTCCC case-control study for Coronary Artery Disease - Combined Controls                 & WTCCC genome-wide case-control association study for Coronary Artery Disease (CAD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.    & NA & Other &  & NA &  TRUE & 1980-01-01T02:00:00+01:00 & FALSE\\\\\n",
       "\t6 & EGAS00000000005 & WTCCC case-control study for Coronary Artery Disease, Hypertension, T2D - combined cases & WTCCC genome-wide case-control association study using the cardiovascular disease CAD, HT and T2D as combined case collection and the 1958 British Birth Cohort and the UK National Blood Service collections as controls. & NA & Other &  & NA &  TRUE & 1980-01-01T02:00:00+01:00 & FALSE\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 10\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | abstract &lt;chr&gt; | study_type &lt;chr&gt; | pubmed_ids &lt;list&gt; | external_links &lt;lgl&gt; | is_released &lt;lgl&gt; | released_date &lt;chr&gt; | is_deprecated &lt;lgl&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | DUMMY_STUDY     | q                                                                                        | NA                                                                                                                                                                                                                         | NA | Other | <!----> | NA | FALSE | 2016-09-26T10:49:02+02:00 | FALSE |\n",
       "| 2 | EGAS00000000001 | WTCCC case-control study for Bipolar Disorder                                            | WTCCC genome-wide case-control association study for Bipolar disorder (BD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                  | NA | Other | <!----> | NA |  TRUE | 1980-01-01T02:00:00+01:00 | FALSE |\n",
       "| 3 | EGAS00000000002 | WTCCC case-control study for Bipolar Disorder - Combined Controls                        | WTCCC genome-wide case-control association study for Bipolar Disorder (BD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.            | NA | Other | <!----> | NA |  TRUE | 1980-01-01T02:00:00+01:00 | FALSE |\n",
       "| 4 | EGAS00000000003 | WTCCC case-control study for Coronary Artery Disease                                     | WTCCC genome-wide case-control association study for Bipolar disorder (CAD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                 | NA | Other | <!----> | NA |  TRUE | 1980-01-01T02:00:00+01:00 | FALSE |\n",
       "| 5 | EGAS00000000004 | WTCCC case-control study for Coronary Artery Disease - Combined Controls                 | WTCCC genome-wide case-control association study for Coronary Artery Disease (CAD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.    | NA | Other | <!----> | NA |  TRUE | 1980-01-01T02:00:00+01:00 | FALSE |\n",
       "| 6 | EGAS00000000005 | WTCCC case-control study for Coronary Artery Disease, Hypertension, T2D - combined cases | WTCCC genome-wide case-control association study using the cardiovascular disease CAD, HT and T2D as combined case collection and the 1958 British Birth Cohort and the UK National Blood Service collections as controls. | NA | Other | <!----> | NA |  TRUE | 1980-01-01T02:00:00+01:00 | FALSE |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id   \n",
       "1 DUMMY_STUDY    \n",
       "2 EGAS00000000001\n",
       "3 EGAS00000000002\n",
       "4 EGAS00000000003\n",
       "5 EGAS00000000004\n",
       "6 EGAS00000000005\n",
       "  title                                                                                   \n",
       "1 q                                                                                       \n",
       "2 WTCCC case-control study for Bipolar Disorder                                           \n",
       "3 WTCCC case-control study for Bipolar Disorder - Combined Controls                       \n",
       "4 WTCCC case-control study for Coronary Artery Disease                                    \n",
       "5 WTCCC case-control study for Coronary Artery Disease - Combined Controls                \n",
       "6 WTCCC case-control study for Coronary Artery Disease, Hypertension, T2D - combined cases\n",
       "  description                                                                                                                                                                                                               \n",
       "1 NA                                                                                                                                                                                                                        \n",
       "2 WTCCC genome-wide case-control association study for Bipolar disorder (BD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                 \n",
       "3 WTCCC genome-wide case-control association study for Bipolar Disorder (BD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.           \n",
       "4 WTCCC genome-wide case-control association study for Bipolar disorder (CAD) using the 1958 British Birth Cohort and the UK National Blood Service collections as controls.                                                \n",
       "5 WTCCC genome-wide case-control association study for Coronary Artery Disease (CAD) using six disease collections together with the 1958 British Birth Cohort and the UK National Blood Service collections as controls.   \n",
       "6 WTCCC genome-wide case-control association study using the cardiovascular disease CAD, HT and T2D as combined case collection and the 1958 British Birth Cohort and the UK National Blood Service collections as controls.\n",
       "  abstract study_type pubmed_ids external_links is_released\n",
       "1 NA       Other                 NA             FALSE      \n",
       "2 NA       Other                 NA              TRUE      \n",
       "3 NA       Other                 NA              TRUE      \n",
       "4 NA       Other                 NA              TRUE      \n",
       "5 NA       Other                 NA              TRUE      \n",
       "6 NA       Other                 NA              TRUE      \n",
       "  released_date             is_deprecated\n",
       "1 2016-09-26T10:49:02+02:00 FALSE        \n",
       "2 1980-01-01T02:00:00+01:00 FALSE        \n",
       "3 1980-01-01T02:00:00+01:00 FALSE        \n",
       "4 1980-01-01T02:00:00+01:00 FALSE        \n",
       "5 1980-01-01T02:00:00+01:00 FALSE        \n",
       "6 1980-01-01T02:00:00+01:00 FALSE        "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>analysis_type</th><th scope=col>genome</th><th scope=col>study_accession_id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>EGAZ00001000000</td><td>REL-2011-01-28 variant calling of Muir samples</td><td>Variants called with samtools mpileup and filtered with varFilter defaults.  Samples were called independently.</td><td>SEQUENCE VARIATION</td><td>GRCh37</td><td>EGAS00001000122</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>EGAZ00001000001</td><td>Whole Genome Sequencing of Neuroblastoma      </td><td>-                                                                                                              </td><td>SEQUENCE VARIATION</td><td>NCBI36</td><td>EGAS00001000222</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>EGAZ00001000002</td><td>Variant calling of UK10K_SCOOP5013826         </td><td>REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          </td><td>SEQUENCE VARIATION</td><td>GRCh37</td><td>EGAS00001000124</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>EGAZ00001000003</td><td>Variant calling of UK10K_SCOOP5013827         </td><td>REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          </td><td>SEQUENCE VARIATION</td><td>GRCh37</td><td>EGAS00001000124</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>EGAZ00001000004</td><td>Variant calling of UK10K_SCOOP5013828         </td><td>REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          </td><td>SEQUENCE VARIATION</td><td>GRCh37</td><td>EGAS00001000124</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>EGAZ00001000005</td><td>Variant calling of UK10K_SCOOP5013829         </td><td>REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          </td><td>SEQUENCE VARIATION</td><td>GRCh37</td><td>EGAS00001000124</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & accession\\_id & title & description & analysis\\_type & genome & study\\_accession\\_id\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & EGAZ00001000000 & REL-2011-01-28 variant calling of Muir samples & Variants called with samtools mpileup and filtered with varFilter defaults.  Samples were called independently. & SEQUENCE VARIATION & GRCh37 & EGAS00001000122\\\\\n",
       "\t2 & EGAZ00001000001 & Whole Genome Sequencing of Neuroblastoma       & -                                                                                                               & SEQUENCE VARIATION & NCBI36 & EGAS00001000222\\\\\n",
       "\t3 & EGAZ00001000002 & Variant calling of UK10K\\_SCOOP5013826          & REL-2011-07-14\\_variant\\_calling\\_of\\_UK10K\\_OBESITY\\_SCOOP                                                           & SEQUENCE VARIATION & GRCh37 & EGAS00001000124\\\\\n",
       "\t4 & EGAZ00001000003 & Variant calling of UK10K\\_SCOOP5013827          & REL-2011-07-14\\_variant\\_calling\\_of\\_UK10K\\_OBESITY\\_SCOOP                                                           & SEQUENCE VARIATION & GRCh37 & EGAS00001000124\\\\\n",
       "\t5 & EGAZ00001000004 & Variant calling of UK10K\\_SCOOP5013828          & REL-2011-07-14\\_variant\\_calling\\_of\\_UK10K\\_OBESITY\\_SCOOP                                                           & SEQUENCE VARIATION & GRCh37 & EGAS00001000124\\\\\n",
       "\t6 & EGAZ00001000005 & Variant calling of UK10K\\_SCOOP5013829          & REL-2011-07-14\\_variant\\_calling\\_of\\_UK10K\\_OBESITY\\_SCOOP                                                           & SEQUENCE VARIATION & GRCh37 & EGAS00001000124\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 6\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | analysis_type &lt;chr&gt; | genome &lt;chr&gt; | study_accession_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | EGAZ00001000000 | REL-2011-01-28 variant calling of Muir samples | Variants called with samtools mpileup and filtered with varFilter defaults.  Samples were called independently. | SEQUENCE VARIATION | GRCh37 | EGAS00001000122 |\n",
       "| 2 | EGAZ00001000001 | Whole Genome Sequencing of Neuroblastoma       | -                                                                                                               | SEQUENCE VARIATION | NCBI36 | EGAS00001000222 |\n",
       "| 3 | EGAZ00001000002 | Variant calling of UK10K_SCOOP5013826          | REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                           | SEQUENCE VARIATION | GRCh37 | EGAS00001000124 |\n",
       "| 4 | EGAZ00001000003 | Variant calling of UK10K_SCOOP5013827          | REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                           | SEQUENCE VARIATION | GRCh37 | EGAS00001000124 |\n",
       "| 5 | EGAZ00001000004 | Variant calling of UK10K_SCOOP5013828          | REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                           | SEQUENCE VARIATION | GRCh37 | EGAS00001000124 |\n",
       "| 6 | EGAZ00001000005 | Variant calling of UK10K_SCOOP5013829          | REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                           | SEQUENCE VARIATION | GRCh37 | EGAS00001000124 |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id    title                                         \n",
       "1 EGAZ00001000000 REL-2011-01-28 variant calling of Muir samples\n",
       "2 EGAZ00001000001 Whole Genome Sequencing of Neuroblastoma      \n",
       "3 EGAZ00001000002 Variant calling of UK10K_SCOOP5013826         \n",
       "4 EGAZ00001000003 Variant calling of UK10K_SCOOP5013827         \n",
       "5 EGAZ00001000004 Variant calling of UK10K_SCOOP5013828         \n",
       "6 EGAZ00001000005 Variant calling of UK10K_SCOOP5013829         \n",
       "  description                                                                                                    \n",
       "1 Variants called with samtools mpileup and filtered with varFilter defaults.  Samples were called independently.\n",
       "2 -                                                                                                              \n",
       "3 REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          \n",
       "4 REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          \n",
       "5 REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          \n",
       "6 REL-2011-07-14_variant_calling_of_UK10K_OBESITY_SCOOP                                                          \n",
       "  analysis_type      genome study_accession_id\n",
       "1 SEQUENCE VARIATION GRCh37 EGAS00001000122   \n",
       "2 SEQUENCE VARIATION NCBI36 EGAS00001000222   \n",
       "3 SEQUENCE VARIATION GRCh37 EGAS00001000124   \n",
       "4 SEQUENCE VARIATION GRCh37 EGAS00001000124   \n",
       "5 SEQUENCE VARIATION GRCh37 EGAS00001000124   \n",
       "6 SEQUENCE VARIATION GRCh37 EGAS00001000124   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>dataset_types</th><th scope=col>technologies</th><th scope=col>num_samples</th><th scope=col>access_type</th><th scope=col>is_in_beacon</th><th scope=col>is_released</th><th scope=col>released_date</th><th scope=col>is_deprecated</th><th scope=col>policy_accession_id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>EGAD00000000001</td><td>WTCCC1 1958BC control dataset</td><td>WTCCC1 project samples from 1958 British Birth Cohort  </td><td></td><td>Affymetr....</td><td>1504</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-11-29T01:00:00+01:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>EGAD00000000002</td><td>NA                           </td><td>WTCCC1 project samples from UK National Blood Service  </td><td></td><td>Affymetr....</td><td>1500</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-11-29T01:00:00+01:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>EGAD00000000003</td><td>NA                           </td><td>WTCCC1 project Bipolar Disorder (BD) samples           </td><td></td><td>            </td><td>   1</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2015-05-27T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>EGAD00000000004</td><td>NA                           </td><td>WTCCC1 project Coronary Artery Disease (CAD) samples   </td><td></td><td>            </td><td>   1</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2015-05-27T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>EGAD00000000005</td><td>NA                           </td><td>WTCCC1 project Inflammatory Bowel Disease (IBD) samples</td><td></td><td>            </td><td>   1</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2015-05-27T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>EGAD00000000006</td><td>NA                           </td><td>WTCCC1 project Hypertension (HT) samples               </td><td></td><td>            </td><td>   1</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2015-05-27T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000391</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & accession\\_id & title & description & dataset\\_types & technologies & num\\_samples & access\\_type & is\\_in\\_beacon & is\\_released & released\\_date & is\\_deprecated & policy\\_accession\\_id\\\\\n",
       "  & <chr> & <chr> & <chr> & <list> & <list> & <int> & <chr> & <lgl> & <lgl> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & EGAD00000000001 & WTCCC1 1958BC control dataset & WTCCC1 project samples from 1958 British Birth Cohort   &  & Affymetr.... & 1504 & controlled & FALSE & TRUE & 2016-11-29T01:00:00+01:00 & FALSE & EGAP00001000391\\\\\n",
       "\t2 & EGAD00000000002 & NA                            & WTCCC1 project samples from UK National Blood Service   &  & Affymetr.... & 1500 & controlled & FALSE & TRUE & 2016-11-29T01:00:00+01:00 & FALSE & EGAP00001000391\\\\\n",
       "\t3 & EGAD00000000003 & NA                            & WTCCC1 project Bipolar Disorder (BD) samples            &  &              &    1 & controlled & FALSE & TRUE & 2015-05-27T02:00:00+02:00 & FALSE & EGAP00001000391\\\\\n",
       "\t4 & EGAD00000000004 & NA                            & WTCCC1 project Coronary Artery Disease (CAD) samples    &  &              &    1 & controlled & FALSE & TRUE & 2015-05-27T02:00:00+02:00 & FALSE & EGAP00001000391\\\\\n",
       "\t5 & EGAD00000000005 & NA                            & WTCCC1 project Inflammatory Bowel Disease (IBD) samples &  &              &    1 & controlled & FALSE & TRUE & 2015-05-27T02:00:00+02:00 & FALSE & EGAP00001000391\\\\\n",
       "\t6 & EGAD00000000006 & NA                            & WTCCC1 project Hypertension (HT) samples                &  &              &    1 & controlled & FALSE & TRUE & 2015-05-27T02:00:00+02:00 & FALSE & EGAP00001000391\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 12\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | dataset_types &lt;list&gt; | technologies &lt;list&gt; | num_samples &lt;int&gt; | access_type &lt;chr&gt; | is_in_beacon &lt;lgl&gt; | is_released &lt;lgl&gt; | released_date &lt;chr&gt; | is_deprecated &lt;lgl&gt; | policy_accession_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | EGAD00000000001 | WTCCC1 1958BC control dataset | WTCCC1 project samples from 1958 British Birth Cohort   | <!----> | Affymetr.... | 1504 | controlled | FALSE | TRUE | 2016-11-29T01:00:00+01:00 | FALSE | EGAP00001000391 |\n",
       "| 2 | EGAD00000000002 | NA                            | WTCCC1 project samples from UK National Blood Service   | <!----> | Affymetr.... | 1500 | controlled | FALSE | TRUE | 2016-11-29T01:00:00+01:00 | FALSE | EGAP00001000391 |\n",
       "| 3 | EGAD00000000003 | NA                            | WTCCC1 project Bipolar Disorder (BD) samples            | <!----> | <!----> |    1 | controlled | FALSE | TRUE | 2015-05-27T02:00:00+02:00 | FALSE | EGAP00001000391 |\n",
       "| 4 | EGAD00000000004 | NA                            | WTCCC1 project Coronary Artery Disease (CAD) samples    | <!----> | <!----> |    1 | controlled | FALSE | TRUE | 2015-05-27T02:00:00+02:00 | FALSE | EGAP00001000391 |\n",
       "| 5 | EGAD00000000005 | NA                            | WTCCC1 project Inflammatory Bowel Disease (IBD) samples | <!----> | <!----> |    1 | controlled | FALSE | TRUE | 2015-05-27T02:00:00+02:00 | FALSE | EGAP00001000391 |\n",
       "| 6 | EGAD00000000006 | NA                            | WTCCC1 project Hypertension (HT) samples                | <!----> | <!----> |    1 | controlled | FALSE | TRUE | 2015-05-27T02:00:00+02:00 | FALSE | EGAP00001000391 |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id    title                        \n",
       "1 EGAD00000000001 WTCCC1 1958BC control dataset\n",
       "2 EGAD00000000002 NA                           \n",
       "3 EGAD00000000003 NA                           \n",
       "4 EGAD00000000004 NA                           \n",
       "5 EGAD00000000005 NA                           \n",
       "6 EGAD00000000006 NA                           \n",
       "  description                                             dataset_types\n",
       "1 WTCCC1 project samples from 1958 British Birth Cohort                \n",
       "2 WTCCC1 project samples from UK National Blood Service                \n",
       "3 WTCCC1 project Bipolar Disorder (BD) samples                         \n",
       "4 WTCCC1 project Coronary Artery Disease (CAD) samples                 \n",
       "5 WTCCC1 project Inflammatory Bowel Disease (IBD) samples              \n",
       "6 WTCCC1 project Hypertension (HT) samples                             \n",
       "  technologies num_samples access_type is_in_beacon is_released\n",
       "1 Affymetr.... 1504        controlled  FALSE        TRUE       \n",
       "2 Affymetr.... 1500        controlled  FALSE        TRUE       \n",
       "3                 1        controlled  FALSE        TRUE       \n",
       "4                 1        controlled  FALSE        TRUE       \n",
       "5                 1        controlled  FALSE        TRUE       \n",
       "6                 1        controlled  FALSE        TRUE       \n",
       "  released_date             is_deprecated policy_accession_id\n",
       "1 2016-11-29T01:00:00+01:00 FALSE         EGAP00001000391    \n",
       "2 2016-11-29T01:00:00+01:00 FALSE         EGAP00001000391    \n",
       "3 2015-05-27T02:00:00+02:00 FALSE         EGAP00001000391    \n",
       "4 2015-05-27T02:00:00+02:00 FALSE         EGAP00001000391    \n",
       "5 2015-05-27T02:00:00+02:00 FALSE         EGAP00001000391    \n",
       "6 2015-05-27T02:00:00+02:00 FALSE         EGAP00001000391    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 13</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>design_description</th><th scope=col>library_name</th><th scope=col>library_construction_protocol</th><th scope=col>paired_nominal_length</th><th scope=col>paired_nominal_sdev</th><th scope=col>platform</th><th scope=col>model</th><th scope=col>library_layout</th><th scope=col>library_strategy</th><th scope=col>library_source</th><th scope=col>library_selection</th><th scope=col>study_accession_id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>EGAX00001000000</td><td>Illumina sequencing of sample \"PD2147b\"; library \"PD2147b_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  </td><td>PD2147b_030809 1  </td><td>Agilent Pulldown</td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>OTHER</td><td>GENOMIC</td><td>Hybrid Selection</td><td>EGAS00001000006</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>EGAX00001000001</td><td>Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b 1\" for the \"Renal Cancer Exome Sequencing\" study.    </td><td>Exp2_PD2126b 1    </td><td>qPCR only       </td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>WGS  </td><td>GENOMIC</td><td>RANDOM          </td><td>EGAS00001000006</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>EGAX00001000002</td><td>Illumina sequencing of sample \"PD2127a\"; library \"Exp2r_PD2127a 1\" for the \"Renal Cancer Exome Sequencing\" study.   </td><td>Exp2r_PD2127a 1   </td><td>qPCR only       </td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>WGS  </td><td>GENOMIC</td><td>RANDOM          </td><td>EGAS00001000006</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>EGAX00001000003</td><td>Illumina sequencing of sample \"PD2147a\"; library \"PD2147a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  </td><td>PD2147a_030809 1  </td><td>Agilent Pulldown</td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>OTHER</td><td>GENOMIC</td><td>Hybrid Selection</td><td>EGAS00001000006</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>EGAX00001000004</td><td>Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b_WGA 1\" for the \"Renal Cancer Exome Sequencing\" study.</td><td>Exp2_PD2126b_WGA 1</td><td>qPCR only       </td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>WGS  </td><td>GENOMIC</td><td>RANDOM          </td><td>EGAS00001000006</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>EGAX00001000005</td><td>Illumina sequencing of sample \"PD3441a\"; library \"PD3441a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  </td><td>PD3441a_030809 1  </td><td>Agilent Pulldown</td><td>250</td><td>NA</td><td>ILLUMINA</td><td>Illumina Genome Analyzer II</td><td>PAIRED</td><td>OTHER</td><td>GENOMIC</td><td>Hybrid Selection</td><td>EGAS00001000006</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 13\n",
       "\\begin{tabular}{r|lllllllllllll}\n",
       "  & accession\\_id & design\\_description & library\\_name & library\\_construction\\_protocol & paired\\_nominal\\_length & paired\\_nominal\\_sdev & platform & model & library\\_layout & library\\_strategy & library\\_source & library\\_selection & study\\_accession\\_id\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <int> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & EGAX00001000000 & Illumina sequencing of sample \"PD2147b\"; library \"PD2147b\\_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   & PD2147b\\_030809 1   & Agilent Pulldown & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & OTHER & GENOMIC & Hybrid Selection & EGAS00001000006\\\\\n",
       "\t2 & EGAX00001000001 & Illumina sequencing of sample \"PD2126b\"; library \"Exp2\\_PD2126b 1\" for the \"Renal Cancer Exome Sequencing\" study.     & Exp2\\_PD2126b 1     & qPCR only        & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & WGS   & GENOMIC & RANDOM           & EGAS00001000006\\\\\n",
       "\t3 & EGAX00001000002 & Illumina sequencing of sample \"PD2127a\"; library \"Exp2r\\_PD2127a 1\" for the \"Renal Cancer Exome Sequencing\" study.    & Exp2r\\_PD2127a 1    & qPCR only        & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & WGS   & GENOMIC & RANDOM           & EGAS00001000006\\\\\n",
       "\t4 & EGAX00001000003 & Illumina sequencing of sample \"PD2147a\"; library \"PD2147a\\_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   & PD2147a\\_030809 1   & Agilent Pulldown & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & OTHER & GENOMIC & Hybrid Selection & EGAS00001000006\\\\\n",
       "\t5 & EGAX00001000004 & Illumina sequencing of sample \"PD2126b\"; library \"Exp2\\_PD2126b\\_WGA 1\" for the \"Renal Cancer Exome Sequencing\" study. & Exp2\\_PD2126b\\_WGA 1 & qPCR only        & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & WGS   & GENOMIC & RANDOM           & EGAS00001000006\\\\\n",
       "\t6 & EGAX00001000005 & Illumina sequencing of sample \"PD3441a\"; library \"PD3441a\\_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   & PD3441a\\_030809 1   & Agilent Pulldown & 250 & NA & ILLUMINA & Illumina Genome Analyzer II & PAIRED & OTHER & GENOMIC & Hybrid Selection & EGAS00001000006\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 13\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | design_description &lt;chr&gt; | library_name &lt;chr&gt; | library_construction_protocol &lt;chr&gt; | paired_nominal_length &lt;int&gt; | paired_nominal_sdev &lt;dbl&gt; | platform &lt;chr&gt; | model &lt;chr&gt; | library_layout &lt;chr&gt; | library_strategy &lt;chr&gt; | library_source &lt;chr&gt; | library_selection &lt;chr&gt; | study_accession_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | EGAX00001000000 | Illumina sequencing of sample \"PD2147b\"; library \"PD2147b_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   | PD2147b_030809 1   | Agilent Pulldown | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | OTHER | GENOMIC | Hybrid Selection | EGAS00001000006 |\n",
       "| 2 | EGAX00001000001 | Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b 1\" for the \"Renal Cancer Exome Sequencing\" study.     | Exp2_PD2126b 1     | qPCR only        | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | WGS   | GENOMIC | RANDOM           | EGAS00001000006 |\n",
       "| 3 | EGAX00001000002 | Illumina sequencing of sample \"PD2127a\"; library \"Exp2r_PD2127a 1\" for the \"Renal Cancer Exome Sequencing\" study.    | Exp2r_PD2127a 1    | qPCR only        | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | WGS   | GENOMIC | RANDOM           | EGAS00001000006 |\n",
       "| 4 | EGAX00001000003 | Illumina sequencing of sample \"PD2147a\"; library \"PD2147a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   | PD2147a_030809 1   | Agilent Pulldown | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | OTHER | GENOMIC | Hybrid Selection | EGAS00001000006 |\n",
       "| 5 | EGAX00001000004 | Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b_WGA 1\" for the \"Renal Cancer Exome Sequencing\" study. | Exp2_PD2126b_WGA 1 | qPCR only        | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | WGS   | GENOMIC | RANDOM           | EGAS00001000006 |\n",
       "| 6 | EGAX00001000005 | Illumina sequencing of sample \"PD3441a\"; library \"PD3441a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.   | PD3441a_030809 1   | Agilent Pulldown | 250 | NA | ILLUMINA | Illumina Genome Analyzer II | PAIRED | OTHER | GENOMIC | Hybrid Selection | EGAS00001000006 |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id   \n",
       "1 EGAX00001000000\n",
       "2 EGAX00001000001\n",
       "3 EGAX00001000002\n",
       "4 EGAX00001000003\n",
       "5 EGAX00001000004\n",
       "6 EGAX00001000005\n",
       "  design_description                                                                                                  \n",
       "1 Illumina sequencing of sample \"PD2147b\"; library \"PD2147b_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  \n",
       "2 Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b 1\" for the \"Renal Cancer Exome Sequencing\" study.    \n",
       "3 Illumina sequencing of sample \"PD2127a\"; library \"Exp2r_PD2127a 1\" for the \"Renal Cancer Exome Sequencing\" study.   \n",
       "4 Illumina sequencing of sample \"PD2147a\"; library \"PD2147a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  \n",
       "5 Illumina sequencing of sample \"PD2126b\"; library \"Exp2_PD2126b_WGA 1\" for the \"Renal Cancer Exome Sequencing\" study.\n",
       "6 Illumina sequencing of sample \"PD3441a\"; library \"PD3441a_030809 1\" for the \"Renal Cancer Exome Sequencing\" study.  \n",
       "  library_name       library_construction_protocol paired_nominal_length\n",
       "1 PD2147b_030809 1   Agilent Pulldown              250                  \n",
       "2 Exp2_PD2126b 1     qPCR only                     250                  \n",
       "3 Exp2r_PD2127a 1    qPCR only                     250                  \n",
       "4 PD2147a_030809 1   Agilent Pulldown              250                  \n",
       "5 Exp2_PD2126b_WGA 1 qPCR only                     250                  \n",
       "6 PD3441a_030809 1   Agilent Pulldown              250                  \n",
       "  paired_nominal_sdev platform model                       library_layout\n",
       "1 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "2 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "3 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "4 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "5 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "6 NA                  ILLUMINA Illumina Genome Analyzer II PAIRED        \n",
       "  library_strategy library_source library_selection study_accession_id\n",
       "1 OTHER            GENOMIC        Hybrid Selection  EGAS00001000006   \n",
       "2 WGS              GENOMIC        RANDOM            EGAS00001000006   \n",
       "3 WGS              GENOMIC        RANDOM            EGAS00001000006   \n",
       "4 OTHER            GENOMIC        Hybrid Selection  EGAS00001000006   \n",
       "5 WGS              GENOMIC        RANDOM            EGAS00001000006   \n",
       "6 OTHER            GENOMIC        Hybrid Selection  EGAS00001000006   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 6</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>biological_sex</th><th scope=col>subject_id</th><th scope=col>phenotype</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>EGAN00000000001</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>EGAN00000000002</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>EGAN00000000003</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>EGAN00000000004</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>EGAN00000000005</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>EGAN00000000006</td><td>NA</td><td>NA</td><td>unknown</td><td>NA</td><td>NA</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 6\n",
       "\\begin{tabular}{r|llllll}\n",
       "  & accession\\_id & title & description & biological\\_sex & subject\\_id & phenotype\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & EGAN00000000001 & NA & NA & unknown & NA & NA\\\\\n",
       "\t2 & EGAN00000000002 & NA & NA & unknown & NA & NA\\\\\n",
       "\t3 & EGAN00000000003 & NA & NA & unknown & NA & NA\\\\\n",
       "\t4 & EGAN00000000004 & NA & NA & unknown & NA & NA\\\\\n",
       "\t5 & EGAN00000000005 & NA & NA & unknown & NA & NA\\\\\n",
       "\t6 & EGAN00000000006 & NA & NA & unknown & NA & NA\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 6\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | biological_sex &lt;chr&gt; | subject_id &lt;chr&gt; | phenotype &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 1 | EGAN00000000001 | NA | NA | unknown | NA | NA |\n",
       "| 2 | EGAN00000000002 | NA | NA | unknown | NA | NA |\n",
       "| 3 | EGAN00000000003 | NA | NA | unknown | NA | NA |\n",
       "| 4 | EGAN00000000004 | NA | NA | unknown | NA | NA |\n",
       "| 5 | EGAN00000000005 | NA | NA | unknown | NA | NA |\n",
       "| 6 | EGAN00000000006 | NA | NA | unknown | NA | NA |\n",
       "\n"
      ],
      "text/plain": [
       "  accession_id    title description biological_sex subject_id phenotype\n",
       "1 EGAN00000000001 NA    NA          unknown        NA         NA       \n",
       "2 EGAN00000000002 NA    NA          unknown        NA         NA       \n",
       "3 EGAN00000000003 NA    NA          unknown        NA         NA       \n",
       "4 EGAN00000000004 NA    NA          unknown        NA         NA       \n",
       "5 EGAN00000000005 NA    NA          unknown        NA         NA       \n",
       "6 EGAN00000000006 NA    NA          unknown        NA         NA       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ega_studies_dt)\n",
    "head(ega_samples_dt)\n",
    "head(ega_experiments_dt)\n",
    "head(ega_analysis_dt)\n",
    "head(ega_datasets_dt)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "0713d2d0-85bc-4c33-bc19-6e1b48bbe91e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 Ã— 12</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>accession_id</th><th scope=col>title</th><th scope=col>description</th><th scope=col>dataset_types</th><th scope=col>technologies</th><th scope=col>num_samples</th><th scope=col>access_type</th><th scope=col>is_in_beacon</th><th scope=col>is_released</th><th scope=col>released_date</th><th scope=col>is_deprecated</th><th scope=col>policy_accession_id</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;list&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;lgl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>838</th><td>EGAD00001000745</td><td>BLUEPRINT: RNA-seq of progenitor cells                                                                   </td><td>Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                           </td><td>Transcri....</td><td>Illumina....</td><td>26</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><th scope=row>996</th><td>EGAD00001000903</td><td>BLUEPRINT release August 2014, RNA-Seq for CD34-negative, CD41-positive, CD42-positive megakaryocyte cell</td><td>RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811</td><td>Transcri....</td><td>Illumina....</td><td> 4</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><th scope=row>1000</th><td>EGAD00001000907</td><td>BLUEPRINT release August 2014, RNA-Seq for common myeloid progenitor                                     </td><td>RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                      </td><td>Transcri....</td><td>Illumina....</td><td> 3</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><th scope=row>1004</th><td>EGAD00001000911</td><td>BLUEPRINT release August 2014, RNA-Seq for erythroblast                                                  </td><td>RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                  </td><td>Transcri....</td><td>Illumina....</td><td> 4</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><th scope=row>1008</th><td>EGAD00001000915</td><td>BLUEPRINT release August 2014, RNA-Seq for megakaryocyte-erythroid progenitor cell                       </td><td>RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                        </td><td>Transcri....</td><td>Illumina....</td><td> 4</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "\t<tr><th scope=row>1011</th><td>EGAD00001000918</td><td>BLUEPRINT release August 2014, RNA-Seq for common lymphoid progenitor                                    </td><td>RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                    </td><td>Transcri....</td><td>Illumina....</td><td> 3</td><td>controlled</td><td>FALSE</td><td>TRUE</td><td>2016-08-17T02:00:00+02:00</td><td>FALSE</td><td>EGAP00001000142</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 Ã— 12\n",
       "\\begin{tabular}{r|llllllllllll}\n",
       "  & accession\\_id & title & description & dataset\\_types & technologies & num\\_samples & access\\_type & is\\_in\\_beacon & is\\_released & released\\_date & is\\_deprecated & policy\\_accession\\_id\\\\\n",
       "  & <chr> & <chr> & <chr> & <list> & <list> & <int> & <chr> & <lgl> & <lgl> & <chr> & <lgl> & <chr>\\\\\n",
       "\\hline\n",
       "\t838 & EGAD00001000745 & BLUEPRINT: RNA-seq of progenitor cells                                                                    & Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                            & Transcri.... & Illumina.... & 26 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t996 & EGAD00001000903 & BLUEPRINT release August 2014, RNA-Seq for CD34-negative, CD41-positive, CD42-positive megakaryocyte cell & RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811 & Transcri.... & Illumina.... &  4 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t1000 & EGAD00001000907 & BLUEPRINT release August 2014, RNA-Seq for common myeloid progenitor                                      & RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                       & Transcri.... & Illumina.... &  3 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t1004 & EGAD00001000911 & BLUEPRINT release August 2014, RNA-Seq for erythroblast                                                   & RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                                   & Transcri.... & Illumina.... &  4 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t1008 & EGAD00001000915 & BLUEPRINT release August 2014, RNA-Seq for megakaryocyte-erythroid progenitor cell                        & RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                         & Transcri.... & Illumina.... &  4 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\t1011 & EGAD00001000918 & BLUEPRINT release August 2014, RNA-Seq for common lymphoid progenitor                                     & RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo\\_sapiens/README\\_rnaseq\\_analysis\\_crg\\_20140811                                     & Transcri.... & Illumina.... &  3 & controlled & FALSE & TRUE & 2016-08-17T02:00:00+02:00 & FALSE & EGAP00001000142\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 Ã— 12\n",
       "\n",
       "| <!--/--> | accession_id &lt;chr&gt; | title &lt;chr&gt; | description &lt;chr&gt; | dataset_types &lt;list&gt; | technologies &lt;list&gt; | num_samples &lt;int&gt; | access_type &lt;chr&gt; | is_in_beacon &lt;lgl&gt; | is_released &lt;lgl&gt; | released_date &lt;chr&gt; | is_deprecated &lt;lgl&gt; | policy_accession_id &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 838 | EGAD00001000745 | BLUEPRINT: RNA-seq of progenitor cells                                                                    | Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                            | Transcri.... | Illumina.... | 26 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| 996 | EGAD00001000903 | BLUEPRINT release August 2014, RNA-Seq for CD34-negative, CD41-positive, CD42-positive megakaryocyte cell | RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811 | Transcri.... | Illumina.... |  4 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| 1000 | EGAD00001000907 | BLUEPRINT release August 2014, RNA-Seq for common myeloid progenitor                                      | RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                       | Transcri.... | Illumina.... |  3 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| 1004 | EGAD00001000911 | BLUEPRINT release August 2014, RNA-Seq for erythroblast                                                   | RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                   | Transcri.... | Illumina.... |  4 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| 1008 | EGAD00001000915 | BLUEPRINT release August 2014, RNA-Seq for megakaryocyte-erythroid progenitor cell                        | RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                         | Transcri.... | Illumina.... |  4 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "| 1011 | EGAD00001000918 | BLUEPRINT release August 2014, RNA-Seq for common lymphoid progenitor                                     | RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                     | Transcri.... | Illumina.... |  3 | controlled | FALSE | TRUE | 2016-08-17T02:00:00+02:00 | FALSE | EGAP00001000142 |\n",
       "\n"
      ],
      "text/plain": [
       "     accession_id   \n",
       "838  EGAD00001000745\n",
       "996  EGAD00001000903\n",
       "1000 EGAD00001000907\n",
       "1004 EGAD00001000911\n",
       "1008 EGAD00001000915\n",
       "1011 EGAD00001000918\n",
       "     title                                                                                                    \n",
       "838  BLUEPRINT: RNA-seq of progenitor cells                                                                   \n",
       "996  BLUEPRINT release August 2014, RNA-Seq for CD34-negative, CD41-positive, CD42-positive megakaryocyte cell\n",
       "1000 BLUEPRINT release August 2014, RNA-Seq for common myeloid progenitor                                     \n",
       "1004 BLUEPRINT release August 2014, RNA-Seq for erythroblast                                                  \n",
       "1008 BLUEPRINT release August 2014, RNA-Seq for megakaryocyte-erythroid progenitor cell                       \n",
       "1011 BLUEPRINT release August 2014, RNA-Seq for common lymphoid progenitor                                    \n",
       "     description                                                                                                                                                                                                                                                                                                                        \n",
       "838  Data supporting the paper Transcriptional diversity during lineage commitment of human blood progenitors                                                                                                                                                                                                                           \n",
       "996  RNA-Seq data for 4 CD34-negative, CD41-positive, CD42-positive megakaryocyte cell sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811\n",
       "1000 RNA-Seq data for 3 common myeloid progenitor sample(s). 3 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                      \n",
       "1004 RNA-Seq data for 4 erythroblast sample(s). 22 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                                  \n",
       "1008 RNA-Seq data for 4 megakaryocyte-erythroid progenitor cell sample(s). 4 run(s), 4 experiment(s), 4 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                        \n",
       "1011 RNA-Seq data for 3 common lymphoid progenitor sample(s). 15 run(s), 3 experiment(s), 3 alignment(s). Part of BLUEPRINT release August 2014. Analysis documentation available at http://ftp.ebi.ac.uk/pub/databases/blueprint/releases/20140811/homo_sapiens/README_rnaseq_analysis_crg_20140811                                    \n",
       "     dataset_types technologies num_samples access_type is_in_beacon\n",
       "838  Transcri....  Illumina.... 26          controlled  FALSE       \n",
       "996  Transcri....  Illumina....  4          controlled  FALSE       \n",
       "1000 Transcri....  Illumina....  3          controlled  FALSE       \n",
       "1004 Transcri....  Illumina....  4          controlled  FALSE       \n",
       "1008 Transcri....  Illumina....  4          controlled  FALSE       \n",
       "1011 Transcri....  Illumina....  3          controlled  FALSE       \n",
       "     is_released released_date             is_deprecated policy_accession_id\n",
       "838  TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "996  TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "1000 TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "1004 TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "1008 TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    \n",
       "1011 TRUE        2016-08-17T02:00:00+02:00 FALSE         EGAP00001000142    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(ega_datasets_dt[ega_datasets_dt$accession_id %in% ega_dt$`Dataset ID`, ])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d6f98382-5625-47e2-aa66-bddd4c47a53e",
   "metadata": {},
   "source": [
    "# backup code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8692c9ce-eafb-48e8-b1ec-4f175e3566f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "  curl https://metadata.ega-archive.org/datasets/{datasetID}/mappings/study_experiment_run_sample \\\n",
    "  -H 'Authorization: Bearer access_token'"
   ]
  },
  {
   "cell_type": "raw",
   "id": "d3a99979-ecb3-4a48-b11f-5a21cce31130",
   "metadata": {},
   "source": [
    "token = 'eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJ3bF9hQkNXY0oyaTJMV0tZVnJFeFpCazBySnBxeGo3YVZ5YURyZC1qRFV3In0.eyJleHAiOjE3MTQwNzQ1NjQsImlhdCI6MTcxNDA3NDI2NCwianRpIjoiMWQxMTRhNTctZmVmMy00NzJlLTg2NDktZGQ0YmRkODkxMGQ1IiwiaXNzIjoiaHR0cHM6Ly9pZHAuZWdhLWFyY2hpdmUub3JnL3JlYWxtcy9FR0EiLCJzdWIiOiJmOjU2YjZkN2U4LTEyYWEtNGM2Ni1iZTE2LThiODVhMTI0ZTJlNTo0NTUwMiIsInR5cCI6IkJlYXJlciIsImF6cCI6Im1ldGFkYXRhLWFwaSIsInNlc3Npb25fc3RhdGUiOiI3MDlkZTk3ZS0wNmRkLTRjZTQtOGQ3Yi00MTJiNzdmZWFjNmUiLCJzY29wZSI6IiIsInNpZCI6IjcwOWRlOTdlLTA2ZGQtNGNlNC04ZDdiLTQxMmI3N2ZlYWM2ZSIsImVnYV9pZCI6NDU1MDJ9.yEWCvrZGx4sbc5gFTKXUlfLpXFIVQ65OqjAOIxC7mBl5inCyny'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "9785c552-0f44-47e5-a6af-1dd80bc93962",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000745/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000745_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000903/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000903_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000907/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000907_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000911/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000911_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000915/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000915_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000918/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000918_ega_meta.json'</span></li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000745/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000745\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000903/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000903\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000907/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000907\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000911/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000911\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000915/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000915\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000918/analyses  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data/EGAD00001000918\\_ega\\_meta.json'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000745/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000745_ega_meta.json'</span>\n",
       "2. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000903/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000903_ega_meta.json'</span>\n",
       "3. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000907/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000907_ega_meta.json'</span>\n",
       "4. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000911/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000911_ega_meta.json'</span>\n",
       "5. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000915/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000915_ega_meta.json'</span>\n",
       "6. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000918/analyses  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data/EGAD00001000918_ega_meta.json'</span>\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000745/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000745_ega_meta.json\"\n",
       "[2] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000903/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000903_ega_meta.json\"\n",
       "[3] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000907/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000907_ega_meta.json\"\n",
       "[4] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000911/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000911_ega_meta.json\"\n",
       "[5] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000915/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000915_ega_meta.json\"\n",
       "[6] \" curl --request GET --url https://metadata.ega-archive.org/datasets/EGAD00001000918/analyses  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data/EGAD00001000918_ega_meta.json\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# download analysis meta file\n",
    "ega_df[, outfile := paste0('data/data/', `Dataset ID`, '_ega_meta.json')]\n",
    "ega_df[, cmd := paste0(\" curl --request GET --url https://metadata.ega-archive.org/datasets/\", `Dataset ID`, \"/analyses \")]\n",
    "ega_df[, cmd := paste0(cmd, \" --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > \", outfile)]\n",
    "head(ega_df$cmd)\n",
    "write(ega_df$cmd, file = 'get_meta.sh')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "55690094-3044-4cc9-86ae-2349dabe486e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000745  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000745_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000903  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000903_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000907  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000907_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000911  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000911_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000915  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000915_ega_meta.json'</span></li><li><span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000918  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000918_ega_meta.json'</span></li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000745  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000745\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000903  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000903\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000907  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000907\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000911  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000911\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000915  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000915\\_ega\\_meta.json'\n",
       "\\item ' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000918  --header \\textbackslash{}'Accept: \\textbackslash{}' --header \\textbackslash{}'Authorization: Bearer undefined\\textbackslash{}' --header \\textbackslash{}'Content-Type: application/json\\textbackslash{}' > data/data2/EGAD00001000918\\_ega\\_meta.json'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000745  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000745_ega_meta.json'</span>\n",
       "2. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000903  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000903_ega_meta.json'</span>\n",
       "3. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000907  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000907_ega_meta.json'</span>\n",
       "4. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000911  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000911_ega_meta.json'</span>\n",
       "5. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000915  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000915_ega_meta.json'</span>\n",
       "6. <span style=white-space:pre-wrap>' curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000918  --header \\'Accept: \\' --header \\'Authorization: Bearer undefined\\' --header \\'Content-Type: application/json\\' &gt; data/data2/EGAD00001000918_ega_meta.json'</span>\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000745  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000745_ega_meta.json\"\n",
       "[2] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000903  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000903_ega_meta.json\"\n",
       "[3] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000907  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000907_ega_meta.json\"\n",
       "[4] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000911  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000911_ega_meta.json\"\n",
       "[5] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000915  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000915_ega_meta.json\"\n",
       "[6] \" curl --request GET --url https://metadata.ega-archive.org/analyses/EGAD00001000918  --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > data/data2/EGAD00001000918_ega_meta.json\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# download analysis meta file\n",
    "ega_df[, outfile2 := paste0('data/data2/', `Dataset ID`, '_ega_meta.json')]\n",
    "ega_df[, cmd := paste0(\" curl --request GET --url https://metadata.ega-archive.org/analyses/\", `Dataset ID`, \" \")]\n",
    "ega_df[, cmd := paste0(cmd, \" --header 'Accept: ' --header 'Authorization: Bearer undefined' --header 'Content-Type: application/json' > \", outfile2)]\n",
    "head(ega_df$cmd)\n",
    "write(ega_df$cmd[1], file = 'get_meta_2.sh')\n",
    "# mappings/analysis_sample\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4bdf6734-092e-4e71-903b-78c0a3792cc2",
   "metadata": {},
   "outputs": [],
   "source": [
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/analyses \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d38084a-7fe7-4651-993f-52a5592c8e2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# sample \n",
    "curl --request GET \\\n",
    "  --url https://metadata.ega-archive.org/datasets/EGAD00001000745/analyses \\\n",
    "  --header 'Accept: ' \\\n",
    "  --header 'Authorization: Bearer undefined' \\\n",
    "  --header 'Content-Type: application/json'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "606ec15b-7a03-47e9-bbbc-648a4e49362f",
   "metadata": {},
   "source": [
    "# read meta file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "8602f12c-13b1-484f-969b-78bb88787dff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/research_jude/rgs01_jude/groups/jxugrp/home/common/Lab_Members/WenhuoHu/collab_Farhan'"
      ],
      "text/latex": [
       "'/research\\_jude/rgs01\\_jude/groups/jxugrp/home/common/Lab\\_Members/WenhuoHu/collab\\_Farhan'"
      ],
      "text/markdown": [
       "'/research_jude/rgs01_jude/groups/jxugrp/home/common/Lab_Members/WenhuoHu/collab_Farhan'"
      ],
      "text/plain": [
       "[1] \"/research_jude/rgs01_jude/groups/jxugrp/home/common/Lab_Members/WenhuoHu/collab_Farhan\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e6851e6-cc6f-4140-a873-c889981f3d39",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "feb24R",
   "language": "R",
   "name": "feb24r"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
